Modules/pyexpat.c

```diff
@@ -103,50 +106,91 @@ struct HandlerInfo {
 
 static struct HandlerInfo handler_info[64];
 
+static int
+set_xml_error_attr_code(PyObject *err, enum XML_Error code)
+{
+    PyObject *py_id_code = PyUnicode_InternFromString("code");
```

Comment:
Ok, it'll be a bit more hard for 3.10 here. What you should do is:

- Add `str_code = PyUnicode_FromString("code");` in `pyexpat_exec` (same construction as for `str_read`).
- Pass the `pyexpat_state *state` to this function to access `code`.

The reason is that, otherwise, you'll need to check that `py_id_code` is not NULL here and if so, return -1. I would prefer interning the string in the module state instead.

Reply:
@picnixz did you mean `state->error`? Pushing a first version now…

Reply:
I meant `str_read` but I think it's only present in my branch. But you implemented what I wanted so it's fine.